core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -75,14 +74,7 @@ sealed trait Chunk[+A] { self =>
 
     if (n <= 0) self
     else if (n >= len) Chunk.empty
-    else
```

Comment:
I thought we were going to keep the old implementations in this PR and do `slice` along with `++`?

Reply:
`private def slice(chunk, start, size)` is a smart constructor for `Slice`, it would make sense to have it since `Slice` changed a bit.

Anyway, went back to the old implementation.